templates/compass.html

```diff
@@ -9,119 +9,160 @@ <h2>Compass<sub class="is-size-7 pl-1">find your way</sub></h2>
     </div>
     <hr>
     <div>
-        <div class="is-flex is-flex-direction-row">
-            <div style="flex:37%;text-align:left;padding:15px;">
-                <h2>Generate Layer</h2>
-                <div id="layerSelectionAdversary">
-                    <select id="layer-selection-adversary" style="margin:0 0 0 0">
-                        <option value="" selected>Select an Adversary (All)</option>
-                        {% for adv in adversaries %}
-                        <option value="{{ adv.adversary_id }}">{{ adv.name }}</option>
-                        {% endfor %}}
-                    </select>
+        <div class="is-flex is-flex-direction-row mb-4">
+            <div class="is-flex is-flex-direction-column">
+                <label for="layer-selection-adversary" class="pb-2 is-size-6">Generate Layer</label>
+                <div class="is-flex is-flex-direction-row is-align-items-center">
+                    <div id="layerSelectionAdversary" class="mt-1 mr-2">
+                        <select id="layer-selection-adversary" x-model="selectedAdversaryID">
+                            <option value="" selected>Select an Adversary (All)</option>
+                            {% for adv in adversaries %}
+                            <option value="{{ adv.adversary_id }}">{{ adv.name }}</option>
+                            {% endfor %}}
+                        </select>
+                    </div>
+                    <label for="generateLayer">
+                        <button id="generateLayer" class="button is-primary is-small"
```

Comment:
We could use a Bulma style here that makes it clear that the button and the select input go together: 

https://bulma.io/documentation/form/general/#form-addons (the first example with the blue search button)